cvat/apps/engine/tests/utils.py

```diff
@@ -125,6 +128,252 @@ def setUp(self):
+
+        response = self._get_request(result_url, user)
+        self.assertEqual(response.status_code, expected_4xx_status_code or status.HTTP_200_OK)
+
+        if not expected_4xx_status_code and file_path:
+            content = BytesIO(b"".join(response.streaming_content))
+            with open(file_path, "wb") as f:
+                f.write(content.getvalue())
+
+        return response
+
+    def _get_api_path_to_export_task_backup(self, task_id: int, *, query_params: dict[str, Any] | None = None) -> str:
+        api_path = f"/api/tasks/{task_id}/backup/export"
+        if query_params:
+            api_path += self._query_params_to_str(**query_params)
+        return api_path
+
+    def _get_api_path_to_export_project_backup(self, project_id: int, *, query_params: dict[str, Any] | None = None) -> str:
+        api_path = f"/api/projects/{project_id}/backup/export"
+        if query_params:
+            api_path += self._query_params_to_str(**query_params)
+        return api_path
+
+    def _get_api_path_to_export_project_dataset(self, project_id: int, *, query_params: dict[str, Any] | None = None) -> str:
+        api_path = f"/api/projects/{project_id}/dataset/export"
+        if query_params:
+            api_path += self._query_params_to_str(**query_params, save_images=True)
+        return api_path
+
+    def _get_api_path_to_export_project_annotations(self, project_id: int, *, query_params: dict[str, Any] | None = None) -> str:
+        api_path = f"/api/projects/{project_id}/dataset/export"
+        if query_params:
+            api_path += self._query_params_to_str(**query_params, save_images=False)
+        return api_path
+
+    def _get_api_path_to_export_task_dataset(self, task_id: int, *, query_params: dict[str, Any] | None = None) -> str:
+        api_path = f"/api/tasks/{task_id}/dataset/export"
+        if query_params:
+            api_path += self._query_params_to_str(**query_params, save_images=True)
+        return api_path
+
+    def _get_api_path_to_export_task_annotations(self, task_id: int, *, query_params: dict[str, Any] | None = None) -> str:
+        api_path = f"/api/tasks/{task_id}/dataset/export"
+        if query_params:
+            api_path += self._query_params_to_str(**query_params, save_images=False)
+        return api_path
+
+    def _get_api_path_to_export_job_dataset(self, job_id: int, *, query_params: dict[str, Any] | None = None) -> str:
+        api_path = f"/api/jobs/{job_id}/dataset/export"
+        if query_params:
+            api_path += self._query_params_to_str(**query_params, save_images=True)
+        return api_path
+
+    def _get_api_path_to_export_job_annotations(self, job_id: int, *, query_params: dict[str, Any] | None = None) -> str:
+        api_path = f"/api/jobs/{job_id}/dataset/export"
+        if query_params:
+            api_path += self._query_params_to_str(**query_params, save_images=False)
+        return api_path
+
```

Comment:
I'm not sure these functions are warranted. Each of them is used exactly once, so you're not reducing any duplication (and IMO, it decreases readability because each function is far away from its caller).

Also, I think it would be better to move the `query_params_to_str` call to `_export`, which _would_ reduce duplication.

Reply:
You are right, thanks